### PR TITLE
Adding crumbs for XSGM

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding test/impl crumbs for XSGM.

## Why is the change being made?

Part of on-going requirement work.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [ ] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/physics/neutronics/tests/test_crossSectionManager.py [hunk 1]
```diff
@@ -172,9 +230,8 @@ def test_createRepresentativeBlock(self):
 
     def test_createRepresentativeBlockDissimilar(self):
         """
-        Test creation of a representative block from a collection with dissimilar blocks
+        Test creation of a representative block from a collection with dissimilar blocks/
```
Comment (albeanth) [inline armi/physics/neutronics/tests/test_crossSectionManager.py:233]:
```suggestion
        Test creation of a representative block from a collection with dissimilar blocks
```

Does ruff not want this all on one line?

File: armi/physics/neutronics/tests/test_crossSectionManager.py [hunk 2]
```diff
@@ -775,6 +831,12 @@ def test_getNextAvailableXsType(self):
         self.assertEqual("D", xsType3)
 
     def test_getRepresentativeBlocks(self):
+        """Test that we can get the representative blocks for a reactor.
```
Comment (albeanth) [inline armi/physics/neutronics/tests/test_crossSectionManager.py:834]:
```suggestion
        """Test that we can create the representative blocks for a reactor.
```

File: armi/physics/neutronics/tests/test_crossSectionManager.py [hunk 3]
```diff
@@ -844,22 +906,37 @@ def test_createRepresentativeBlocksUsingExistingBlocks(self):
...
         self.assertTrue(self.csm.representativeBlocks)
 
     def test_interactEveryNode(self):
-        """Test `everyNode` lattice physics update frequency."""
+        """Test `everyNode` lattice physics update frequency.
+
+        .. test:: The XSGM frequency depends on the LPI frequency at every time node.
+            :id: T_ARMI_XSGM_FREQ2
+            :tests: R_ARMI_XSGM_FREQ
+        """
```
Comment (albeanth) [inline armi/physics/neutronics/tests/test_crossSectionManager.py:939]: Can we put this test tag on the two coupled tests below too? Thanks!

File: armi/physics/neutronics/tests/test_crossSectionManager.py [hunk 4]
```diff
@@ -93,11 +94,69 @@ def setUp(self):
         self.bc.extend(self.blockList)
 
     def test_createRepresentativeBlock(self):
-
         avgB = self.bc.createRepresentativeBlock()
         self.assertAlmostEqual(avgB.p.percentBu, 50.0)
 
 
+class TestCylindricalComponentsAverageBlockCollection(unittest.TestCase):
```
Comment (albeanth) [inline armi/physics/neutronics/tests/test_crossSectionManager.py:101]: `CylindricalComponentsAverageBlockCollection::_makeRepresentativeBlock` has coverage already in `TestBlockCollectionComponentAverage1DCylinder::test_ComponentAverage1DCylinder` so this test may not be needed. It's just buried in a more "in-repo integration test" style. 

Thoughts?

Comment (john-science) [reply armi/physics/neutronics/tests/test_crossSectionManager.py]: sure. Done.

File: armi/physics/neutronics/crossSectionGroupManager.py [hunk 5]
```diff
@@ -832,6 +838,12 @@ def __init__(self, r, cs):
         self._unrepresentedXSIDs = []
 
     def interactBOL(self):
+        """Called at the Beginning-of-Life of a run, before any cycles start.
+
+        .. impl:: The lattice physics interface and XSGM are connected in when they run.
+            :id: I_ARMI_XSGM_FREQ
+            :implements: R_ARMI_XSGM_FREQ
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/physics/neutronics/crossSectionGroupManager.py:845]: I mean... sure.  There are no extra points for having MORE, but if you like. Done.

File: armi/physics/neutronics/tests/test_crossSectionManager.py (hunk 3 above)
Comment (john-science) [reply armi/physics/neutronics/tests/test_crossSectionManager.py:883]: Done

File: armi/physics/neutronics/crossSectionGroupManager.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (albeanth) [reply armi/physics/neutronics/crossSectionGroupManager.py:845]: My thought process here is that we don't want people fussing with the `interact` methods where the latticePhysicsFrequency is called. Slapping the `impl` tag on it creates a nice barrier/stop check for this.

**Review by albeanth — APPROVED** [review]

**Pull request merged**